server/manager/src/main/java/org/apache/accumulo/manager/tableOps/compact/CleanUp.java

```diff
@@ -74,16 +75,21 @@ public long isReady(FateId fateId, Manager manager) throws Exception {
 
     try (
         var tablets = ample.readTablets().forTable(tableId).overlapping(startRow, endRow)
-            .fetch(PREV_ROW, COMPACTED).checkConsistency().build();
+            .fetch(PREV_ROW, COMPACTED, USER_COMPACTION_REQUESTED).checkConsistency().build();
         var tabletsMutator = ample.conditionallyMutateTablets(resultConsumer)) {
 
       t1 = System.nanoTime();
       for (TabletMetadata tablet : tablets) {
         total++;
         if (tablet.getCompacted().contains(fateId)) {
-          tabletsMutator.mutateTablet(tablet.getExtent()).requireAbsentOperation()
-              .requireSame(tablet, COMPACTED).deleteCompacted(fateId)
-              .submit(tabletMetadata -> !tabletMetadata.getCompacted().contains(fateId));
+          var mutator = tabletsMutator.mutateTablet(tablet.getExtent()).requireAbsentOperation()
+              .requireSame(tablet, COMPACTED).deleteCompacted(fateId);
+          if (tablet.getUserCompactionsRequested().contains(fateId)) {
+            mutator.deleteUserCompactionRequested(fateId);
+          }
+
+          mutator.submit(tabletMetadata -> !tabletMetadata.getCompacted().contains(fateId)
+              && !tabletMetadata.getUserCompactionsRequested().contains(fateId));
           submitted++;
         }
```

Comment:
Was wondering about the case where user compaction requested column exists but the compacted column does not.  Does not seem like this should happen, was wondering if we should throw an error.  

```suggestion
        } else if(tablet.getUserCompactionsRequested().contains(fateId)){ 
            throw new IllegalStateException("saw user comp req but did not see compacted for <tablet> <fateId>");
       }
```

Reply:
This was refactored to make the checks independent (still one mutation)